GUDHI_SUBSAMPLING_USE_CGAL & Eigen

For GUDHI_SUBSAMPLING_USE_CGAL, we only check if we have a recent enough version of CGAL. However, we use Epick_d, so we need Eigen. We could probably test `__has_include(<Eigen/Jacobi>)` to match the check in CGAL.

It would be easy to write a mini-kernel for this use, but the case where we have CGAL and not Eigen is not interesting.

This issue was detected on [Azure Windows CI](https://dev.azure.com/GUDHI/gudhi-devel/_build/results?buildId=2344&view=logs&jobId=2d2b3007-3c5c-5840-9bb0-2b1ea49925f3&j=2d2b3007-3c5c-5840-9bb0-2b1ea49925f3&t=f33089cd-d3fe-578c-2c5e-4ee560e10a0d) and [pip build Windows CI](https://github.com/GUDHI/gudhi-devel/actions/runs/5488541788/job/14864455012):
```
C:/vcpkg/installed/x64-windows/include\CGAL/NewKernel_d/Cartesian_LA_base.h(31): fatal error C1189: #error:  Eigen3 is required
error: command 'C:\\Program Files\\Microsoft Visual Studio\\2022\\Enterprise\\VC\\Tools\\MSVC\\14.35.32215\\bin\\HostX86\\x64\\cl.exe' failed with exit status 2
```
because vcpkg failed to install Eigen3:
```
Building eigen3:x64-windows...
-- Downloading https://gitlab.com/libeigen/eigen/-/archive/3.4.0/eigen-3.4.0.tar.gz -> libeigen-eigen-3.4.0.tar.gz...
[DEBUG] To include the environment variables in debug output, pass --debug-env
[DEBUG] Trying to load bundleconfig from C:\vcpkg\vcpkg-bundle.json
[DEBUG] Failed to open: C:\vcpkg\vcpkg-bundle.json
[DEBUG] Bundle config: readonly=false, usegitregistry=false, embeddedsha=nullopt, deployment=Git, vsversion=nullopt
[DEBUG] Metrics enabled.
[DEBUG] Feature flag 'binarycaching' unset
[DEBUG] Feature flag 'compilertracking' unset
[DEBUG] Feature flag 'registries' unset
[DEBUG] Feature flag 'versions' unset
[DEBUG] Feature flag 'dependencygraph' unset
Downloading https://gitlab.com/libeigen/eigen/-/archive/3.4.0/eigen-3.4.0.tar.gz
error: Failed to download from mirror set
error: https://gitlab.com/libeigen/eigen/-/archive/3.4.0/eigen-3.4.0.tar.gz: failed: status code 503
```
CMake detects it:
```
 -- CGAL version: 5.5.2.
-- Multi-configuration CMake generator: cannot display flags
-- Could NOT find TBB (missing: TBB_DIR)
-- Could NOT find Eigen3 (missing: EIGEN3_INCLUDE_DIR) (Required is at least version "3.1.0")
```

---

The strange part of this issue is that it works fine for ["python without cgal" Linux CircleCI](https://github.com/GUDHI/gudhi-devel/blob/4184619d091ea3f0737a44dcead61472810befa6/.circleci/config.yml#L291-L309)

--- a/src/python/gudhi/subsampling.pyx
+++ b/src/python/gudhi/subsampling.pyx
@@ -130,7 +130,7 @@ def sparsify_point_set(points=None, off_file='', min_squared_dist=0.0):
     :rtype: List[List[float]]
     """
     if not GUDHI_SUBSAMPLING_USE_CGAL:
-        raise NotImplementedError("subsampling sparsify_point_set is only available for CGAL >= 4.11")
+        raise NotImplementedError("subsampling sparsify_point_set is only available with CGAL >= 4.11 and Eigen3")
     
     if off_file:
         if os.path.isfile(off_file):

--- a/src/python/include/Subsampling_interface.h
+++ b/src/python/include/Subsampling_interface.h
@@ -19,7 +19,7 @@
 // Default value is undefined
 #define _GUDHI_SUBSAMPLING_USE_CGAL 0
 
-#if __has_include(<CGAL/version.h>)
+#if __has_include(<CGAL/version.h>) && __has_include(<Eigen/Jacobi>)
 #include <CGAL/version.h>  // for CGAL_VERSION_NR
 #if CGAL_VERSION_NR >= 1041101000
 #undef _GUDHI_SUBSAMPLING_USE_CGAL  // To remove warning

